TransIT.DAL/Repositories/BaseRepository.cs

```diff
@@ -17,53 +17,54 @@ public abstract class BaseRepository<TEntity> : IBaseRepository<TEntity>, IQuery
 
         public BaseRepository(TransITDBContext context)
         {
+            _entities = context.Set<TEntity>();
             _context = context;
         }
 
         public virtual Task<TEntity> GetByIdAsync(int id)
         {
-            return ComplexEntities.SingleOrDefaultAsync(t => t.Id.Equals(id));
+            return ComplexEntities.SingleOrDefaultAsync(entity => entity.Id == id);
         }
 
         public virtual Task<IEnumerable<TEntity>> GetAllAsync()
         {
-            return Task.FromResult<IEnumerable<TEntity>>(ComplexEntities);
+            return Task.FromResult(ComplexEntities.AsEnumerable());
         }
 
-        public virtual Task<IEnumerable<TEntity>> GetAllAsync(Expression<Func<TEntity, bool>> predicate)
+        public async virtual Task<IEnumerable<TEntity>> GetAllAsync(Expression<Func<TEntity, bool>> predicate)
```

Comment:
remove async await keywords, just keywords

Reply:
Because , when i remove await, it cannot convert Task<List<item> to task Task<IEnumerable<item>>